pgsync/querybuilder.py

```diff
@@ -719,10 +719,16 @@ def _non_through(self, node: Node) -> None:  # noqa: C901
             onclause: list = []
 
             foreign_keys: dict = self._get_foreign_keys(node, child)
-
+            table: Optional[str] = (
+                child.relationship.throughs[0].table
+                if child.relationship.throughs
+                else None
+            )
             foreign_key_columns: list = self._get_column_foreign_keys(
                 child._subquery.columns,
                 foreign_keys,
+                table=table,
+                schema=node.schema,
```

Comment:
looks good to me. only thing I would suggest is to use child.schema here because we do allow cross schema queries. So the child could be in a different namespace.